src/main/java/org/opensearch/security/dlic/rest/api/ActionGroupsApiAction.java

```diff
@@ -186,7 +212,7 @@ public Map<String, RequestContentValidator.DataType> allowedKeys() {
 
                     @Override
                     public Set<String> mandatoryKeys() {
-                        return ImmutableSet.of("allowed_actions");
+                        return ImmutableSet.of("allowed_actions", "type");
```

Comment:
this would break existing clients, right? I think we should not add new mandatory keys in minor versions if we target this fix to be released in next version, thoughts?

Reply:
True, I will add a warning message in case any action group does not have the `type` property.

Reply:
+1 to a warning message. When creating custom action groups through security-dashboards-plugin it doesn't assign a type. Its technically possible to create an action groups that has transport action names that are a mixture of cluster permissions and index permissions. IMO we should improve the experience in the future because it doesn't make sense to have a mixture in the same action group.

Reply:
I added it already :-)

Reply:
Thank you!